coinswap-react-native/android/src/main/java/org/coinswap/reactnative/CoinswapModule.kt

```
@@ -0,0 +1,256 @@
+    }
+  }
+
+  private fun requireTaker(takerId: String): Taker {
+    return takers[takerId] ?: throw IllegalArgumentException("Unknown taker id: $takerId")
+  }
+
+  private fun launchPromise(promise: Promise, block: () -> Any?) {
+    scope.launch {
+      try {
+        promise.resolve(block())
+      } catch (error: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)